go/proof_test.go

```diff
@@ -13,50 +13,50 @@ func TestExistenceProof(t *testing.T) {
 	}{
 		"must have at least one step": {
 			proof: &ExistenceProof{
-				Key:   []byte("foo"),
-				Value: []byte("bar"),
+				Key:       []byte("foo"),
+				ValueHash: []byte("bar"),
 			},
 			isErr: true,
 		},
 		// copied from ops_test / TestLeafOp
 		"executes one leaf step": {
 			proof: &ExistenceProof{
-				Key:   []byte("food"),
-				Value: []byte("some longer text"),
+				Key:       []byte("food"),
+				ValueHash: []byte("some longer text"),
 				Leaf: &LeafOp{
 					Hash:   HashOp_SHA256,
 					Length: LengthOp_VAR_PROTO,
 				},
 			},
-			expected: fromHex("b68f5d298e915ae1753dd333da1f9cf605411a5f2e12516be6758f365e6db265"),
+			expected: fromHex("7496fdaa49e3764d635f9f21e60cec0a75b8d7595a9a3bb013692bb45d14e326"),
 		},
 		// iavl leaf: start with 0, length 3
 		// inner prefix: !start with 0, length >= 4
 		"demonstrate maliability of leaf if we change leaf algorithm": {
 			proof: &ExistenceProof{
-				Key:   append([]byte{4}, []byte("food")...),
-				Value: append([]byte{16}, []byte("some longer text")...),
+				Key:       append([]byte{4}, []byte("food")...),
+				ValueHash: []byte("some longer text"),
```

Comment:
It would be best to keep the hashes all the same in these tests, so we see this doesn't change the proof format, just the metadata passed along with the proof. (Here I refer to leaf+inner as the proof, and key+value (maybe hashed?) as the metadata).

If you just use `ValueHash: append([]byte{16}, []byte("some longer text")...),` you should get the same expected result (as this would be using `PrehashKey` and `PrehashValue` as `NoHash` - https://github.com/confio/ics23/blob/master/proofs.proto#L109-L117)